src/pages/Chat/Chat/index.tsx

```diff
@@ -38,30 +38,68 @@ export default (): JSX.Element => {
   const [messages, setMessages] = useState<IMessage[]>([]);
   const [isLoadingEarlier, setIsLoadingEarlier] = useState<boolean>(false);
   const [chatRef, setChatRef] = useState<DocumentRefData>();
+  const [otherIsTyping, setOtherIsTyping] = useState<boolean>(false);
+  const typingTimeoutFunction = useRef<NodeJS.Timeout>();
+  const ignoreFirstOnInputTextChanged = useRef<boolean>(true);
+  const selfIsTyping = useRef<boolean>(false);
+  const firstLoad = useRef<boolean>(true);
 
   const sorter = (a: IMessage, b: IMessage): number => (b.createdAt as Date).getTime() - (a.createdAt as Date).getTime();
 
+  const naiveUnique = (msgs: IMessage[]): IMessage[] => {
```

Comment:
Is this method necessary for the final version? It seems to be here to prevent duplicate messages from existing and to warn the user when that happens. I can see it being useful for initial tests (and I will probably want it for when I test everything tonight), but shouldn't we remove it after the final tests?

Reply:
It is necessary. There are many sources to update messages and they are asynchronous.